ogr/ogrsf_frmts/odbc/ogrodbclayer.cpp

```diff
@@ -260,17 +260,36 @@ OGRFeature *OGRODBCLayer::GetNextRawFeature()
 /* -------------------------------------------------------------------- */
     for( int iField = 0; iField < poFeatureDefn->GetFieldCount(); iField++ )
     {
+        const OGRFieldType eType = poFeatureDefn->GetFieldDefn(iField)->GetType();
         int iSrcField = panFieldOrdinals[iField]-1;
-        const char *pszValue = poStmt->GetColData( iSrcField );
-
-        if( pszValue == nullptr )
-            poFeature->SetFieldNull( iField );
-        else if( poFeature->GetFieldDefnRef(iField)->GetType() == OFTBinary )
-            poFeature->SetField( iField,
-                                 poStmt->GetColDataLength(iSrcField),
-                                 (GByte *) pszValue );
+
+        if ( eType == OFTReal && poStmt->Flags() & CPLODBCStatement::Flag::RetrieveNumericColumnsAsDouble )
```

Comment:
just to avoid relying on operator precedence rules:
```suggestion
        if ( eType == OFTReal && (poStmt->Flags() & CPLODBCStatement::Flag::RetrieveNumericColumnsAsDouble) )
```